### DISCUSSION TITLE
Failed to load url ../.vscode/extensions/vitest.explorer-0.5.15/dist/setupFile.mjs even though that file exists?

### DISCUSSION THREAD

**Discussion by rhefner** [vitest-dev/vscode#337] [Q&A] [answered]
Updating to the pre-release v0.5.15 I get the following when using in-source testing:

![image](https://github.com/vitest-dev/vscode/assets/8144799/3f8a8282-a355-476a-b79d-6bcd37c6a008)

But the file exists:

```
$ ls -ltrah /Users/richard.hefner/.vscode/extensions/vitest.explorer-0.5.15/dist/setupFile.mjs
-rw-r--r--  1 richard.hefner  staff   203B Apr 11 11:06 /Users/richard.hefner/.vscode/extensions/vitest.explorer-0.5.15/dist/setupFile.mjs
```

The in-source test is similar to:

```
if (import.meta.vitest) {
  const { render, screen } = await import('@testing-library/react')
  const { createRemixStub } = await import('@remix-run/testing')
  const { ErrorBoundaryThatThrows } = await import('@myorg/remix/test-utils')

  let App: ReturnType<typeof createRemixStub>

  beforeEach(() => {
    App = createRemixStub([
      {
        path: '/',
        Component: Index,
        ErrorBoundary: ErrorBoundaryThatThrows,
      },
    ])
  })

  describe('home', () => {
    test('renders home page', () => {
      render(<App initialEntries={['/']} />)
    })
  })
}
```

When attempting to run the file, nothing is output in the "Vitest" output channel. Running the test manually with vitest works as expected.

Anyone else observed this?

**Comment by rhefner** [comment]
Hmm... This seems to happen even moving the test out into it's own file. :-\

**Comment by sheremet-va** [comment, marked as answer]
Just to test a theory, does adding `server.fs.strict: false` to the config helps?

```ts
export default defineConfig({
  server: {
    fs: {
      strict: false,
    }
  }
})
```

**Reply by rhefner** [reply]
Your _theory_ was the fix for this issue. Thank you, @sheremet-va!

**Reply by sheremet-va** [reply]
Please, don't disable it just to fix vscode issue. I will release a patch where we do something similar in a more secure way. In the meantime, you can safely disable it if `process.env.VITEST_VSCODE` is present for example.

**Reply by rhefner** [reply]
Understood.  👍🏻

**Reply by herrKlein** [reply]
I was banging my head against the wall for this issue. 
setting this resolved the error. 
```
server: {
    fs: {
      strict: process.env.VITEST_VSCODE ? false : true,
    }
  }
```
But I would like to know if it's patched, and if there is a more descent fix now to not get the error :
```Failed to load url ../.vscode/extensions/vitest.explorer-0.5.15/dist/setupFile.mjs```

**Reply by sheremet-va** [reply]
There should be no such errors in the latest versions.

**Reply by PPDEMOone** [reply]
I have the same problem, but it's not working for me. :(
